### PR TITLE
More New API changes for MapChooser, Nominations, RockTheVote, RandomCycle, and NextMap

### PR DESCRIPTION
**Pull request by powerlord**
This depends on [pull request 225](https://github.com/alliedmodders/sourcemod/pull/225) and mapchooser will not compile without it.

Changes include things like:

* Converting old style variable declaration to new style
* Converting old-style casts to view_as
* Converting from Handle to ArrayList where adt_arrays are being used.

There are also misc cleanups that were missed prior to this:

* Converting hard-coded iength for map names to be PLATFORM_MAX_PATH instead of 32, 33, or 64.
* Converting hard-coded lengths for client names to MAX_NAME_LENGTH+1 (yay null terminator).

Note: This is only the first upgrade I have planned for MapChooser.  I also have plans for introducing fixes for various issues with different games, but those need more testing.

### CODE REVIEW

**Comment by powerlord** [comment]
Side note:  Since this hasn't been mentioned yet, this pull request also fixes a few locations that were still hard-coding 32, 64, or 65 for the map length.

**Comment by psychonic** [comment]
Whoops! I completely forgot about this PR.

> Side note: Since this hasn't been mentioned yet, this pull request also fixes a few locations that were still hard-coding 32, 64, or 65 for the map name string.

I took care of this yesterday in #343, landing shortly, but there shouldn't be a conflict since both use PLATFORM_MAX_PATH for the sizes now.

I do see a couple of error reported on this though, due to the added "#pragma newdecls required":
> C:\projects\sourcemod\plugins\mapchooser.sp(562) : error 240: 'MenuAction:' is an old-style tag operation; use view_as<MenuAction>(expression) instead
> C:\projects\sourcemod\plugins\mapchooser.sp(794) : error 240: 'MenuAction:' is an old-style tag operation; use view_as<MenuAction>(expression) instead
 
Do you mind fixing the errors and merging master back into your branch since it's been quite some time (our fault)?

**Comment by powerlord** [comment]
@psychonic The bugs you're seeing are actually a bug in menus.inc and were fixed in master by PR #229.

**Comment by psychonic** [comment]
Touché.

**Comment by powerlord** [comment]
Urgh, attempting to merge master into this is introducing all sorts of... fun... conflicts.

**Comment by powerlord** [comment]
I seriously hope this didn't mess up the sourcepawn submodule.  I had to Google how to resolve it.

**Comment by powerlord** [comment]
*sigh* of course it f'ed it up.  Time to rollback and try again.

File: plugins/include/mapchooser.inc [hunk 1]
```diff
@@ -51,7 +82,7 @@ native bool:RemoveNominationByOwner(owner);
  * @param array		An ADT array handle to add the map strings to.
  * @noreturn
```
Comment (psychonic) [inline plugins/include/mapchooser.inc]: You can ditch these noreturn tags for void returns if you want. They were necessary before it was clear whether a call even returned a value or not.

**Comment by psychonic** [comment]
All of the newdecl and buffer size updates look good. I still need to review the actual functionality changes, which is fortunately only a small minority of this.

**Comment by psychonic** [comment]
I misread the PR and some of the code I guess. "New API" is referring to the new-style syntax and methodmaps, not new mapchooser API. I thought that I had spotted some changes to the nominations API. Will take another look.

**Comment by psychonic** [comment]
>  I thought that I had spotted some changes to the nominations API. Will take another look.

It was the comments on the native impls that were throwing me off, making it look like changes that were really just updating a comment to match the prototype in the inc.

All looks good. Thanks for updating these.

**Pull request merged**